allennlp/data/iterators/multiprocess_iterator.py

```diff
@@ -0,0 +1,140 @@
+from typing import Iterable, Iterator, List, Optional
+import copy
+import logging
+
+from torch.multiprocessing import Process, Queue, get_logger
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.data.instance import Instance
+from allennlp.data.iterators.data_iterator import DataIterator, TensorDict
+from allennlp.data.dataset import Batch
+from allennlp.data.vocabulary import Vocabulary
+
+logger = get_logger()  # pylint: disable=invalid-name
+logger.setLevel(logging.INFO)
+
+def _create_tensor_dicts(input_queue: Queue,
+                         output_queue: Queue,
+                         iterator: DataIterator,
+                         index: int) -> None:
+    """
+    Pulls at most ``max_instances_in_memory`` from the input_queue,
+    groups them into batches of size ``batch_size``, converts them
+    to ``TensorDict`` s, and puts them on the ``output_queue``.
+    """
+    def instances() -> Iterator[Instance]:
+        instance = input_queue.get()
+        while instance is not None:
+            yield instance
+            instance = input_queue.get()
+
+    for tensor_dict in iterator(instances(), num_epochs=1, shuffle=False):
```

Comment:
Shouldn't we be passing down the `shuffle` parameter here? I think it will ignore whatever `shuffle` parameter is passed to `__call__` without it.